src/Compilers/CSharp/Portable/SourceGeneration/CSharpSyntaxHelper.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.SourceGeneration;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed class CSharpSyntaxHelper : AbstractSyntaxHelper
```

Comment:
similar to ISyntaxFacts in IDE.  This allows us to have algorithms in the compiler layer that abstract over VB/C#.